src/meta/processors/job/JobManager.cpp

```diff
@@ -201,19 +214,24 @@ void JobManager::cleanJob(JobID jobId) {
   }
 }
 
-nebula::cpp2::ErrorCode JobManager::jobFinished(JobID jobId, cpp2::JobStatus jobStatus) {
-  LOG(INFO) << folly::sformat(
-      "{}, jobId={}, result={}", __func__, jobId, apache::thrift::util::enumNameSafe(jobStatus));
+nebula::cpp2::ErrorCode JobManager::jobFinished(GraphSpaceID spaceId,
+                                                JobID jobId,
+                                                cpp2::JobStatus jobStatus) {
+  LOG(INFO) << folly::sformat("{}, spaceId={}, jobId={}, result={}",
+                              __func__,
+                              spaceId,
+                              jobId,
+                              apache::thrift::util::enumNameSafe(jobStatus));
   // normal job finish may race to job stop
-  std::lock_guard<std::recursive_mutex> lk(muJobFinished_);
-  auto optJobDescRet = JobDescription::loadJobDescription(jobId, kvStore_);
+  std::lock_guard<std::recursive_mutex> lk(muJobFinished_[spaceId]);
```

Comment:
The `recursive_mutex` is related to the `setFinishCallback`, you could ask @liwenhui-soul why we need it.

Reply:
muJobFinished_  used to mutually exclusive `start & finish & stop a job`

Reply:
I mean why we need `recursive_mutex` instead of `mutex`

Reply:
To prevent deadlocks when nesting calls. For example, `runJobInternal` calls `jobFinished`, and locks both `runJobInternal` and `jobFinished`.

Reply:
👌